test/uk/gov/hmrc/entrydeclarationdecision/services/ProcessDecisionServiceSpec.scala

```diff
@@ -233,30 +239,32 @@ class ProcessDecisionServiceSpec
     xmlBuilderMock: (Decision[R], E) => CallHandler[Elem]): Unit = {
     val acceptance = messageType.isAcceptance
 
-    "enrich, build xml and send to outcome" in new Test {
-      MockAppConfig.validateJsonToXMLTransformation returns false
+    def setupMocks(validateJsonToXMLTransformation: Boolean, longJourneyTime: FiniteDuration) = {
+      MockAppConfig.validateJsonToXMLTransformation returns validateJsonToXMLTransformation
+      MockAppConfig.longJourneyTime returns longJourneyTime
       enrichmentConnectorMock(submissionId) returns Right(enrichment)
-
       xmlBuilderMock(decision, enrichment) returns rawXml
       MockXMLWrapper.wrapXml(correlationId, rawXml) returns wrappedXml
       MockOutcomeConnector.send(validOutcome(messageType, acceptance)) returns Future.successful(Right(()))
       MockStoreConnector.setShortTtl(submissionId) returns Future.successful(true)
+    }
 
+    "enrich, build xml and send to outcome" in new Test {
+      setupMocks(validateJsonToXMLTransformation = false, longJourneyTime)
       service.processDecision(decision).futureValue shouldBe Right(())
 
       shouldReportMetric()
     }
+    "log for long journey times" in new Test {
+      setupMocks(validateJsonToXMLTransformation = false, journeyTime)
+      service.processDecision(decision).futureValue shouldBe Right(())
 
+      shouldReportMetric()
+      MockPagerDutyLogger.logLongJourneyTime(journeyTime, journeyTime) returns Unit
```

Comment:
Minor as it only affects the log message, but would it be possible to have a different value for the two parameters to check that they are passed in the right order..? (But see comment on mock.)